core/math/a_star_grid_2d.cpp

```diff
@@ -734,6 +734,119 @@ TypedArray<Vector2i> AStarGrid2D::get_id_path(const Vector2i &p_from_id, const V
 	return path;
 }
 
+Vector2 AStarGrid2D::get_flow_direction(const Vector2i &p_id) const {
+	ERR_FAIL_COND_V_MSG(dirty, Vector2(), "Grid is not initialized. Call the update method.");
+	ERR_FAIL_COND_V_MSG(!is_in_boundsv(p_id), Vector2(), vformat("Can't get point's flow direction. Point %s out of bounds %s.", p_id, region));
+	return _get_point_unchecked(p_id)->flow_direction;
+}
+
+TypedArray<Vector2> AStarGrid2D::get_flow_directions(const Rect2i &p_region) const {
+	ERR_FAIL_COND_V_MSG(dirty, TypedArray<Vector2i>(), "Grid is not initialized. Call the update method.");
+	ERR_FAIL_COND_V_MSG(!region.encloses(p_region), TypedArray<Vector2i>(), vformat("Can't get point's flow direction. Region %s out of bounds %s.", p_region, region));
+
+	TypedArray<Vector2> result;
+	result.resize(p_region.size.x * p_region.size.y);
+
+	uint32_t index = 0U;
+	for (int y = p_region.get_position().y; y < p_region.get_end().y; y++) {
```

Comment:
If you stored the Vector2 flow_direction in a parrallel array instead of directly in the point, then all of this can just become a `memcpy` which will be faster because it will be vectorized.

Reply:
See, its not possible to setup that array otherwise:

![изображение](https://github.com/user-attachments/assets/0e9b90f5-5204-4b43-b729-d82fdad7ef69)

Reply:
These were just random thoughts on how to speed up things, but you are right that it is not possible right now.  You can just close this.